### PR TITLE
Add nicer error handling

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?

Adds the `anyhow` crate, which makes it much easier to handle errors.

### Motivation

The custom error and all the `.unwrap()`s were getting out of hand.

### Additional Notes

This PR depend on #18, which is why you see the cachegrind commits.

### Checklist

[] I have added tests for the code I am submitting
[] My unit tests cover both failure and success scenarios
[] If applicable, I have discussed my architecture

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -121,10 +122,9 @@ async fn run_setup(setup: &[String], env: &HashMap<String, String>) {
             .stdout(get_stdio())
             .stderr(get_stdio())
             .status()
-            .await
-            .unwrap()
+            .await?
             .code()
-            .unwrap();
+            .ok_or(anyhow!("no exist code"))?;
```
Comment (bengl) [inline src/main.rs:127]: typo. will fix.

**Comment by brettlangdon** [comment]
I can't tell what from here depends on #18

**Review by pawelchcki — DISMISSED** [review]
👍 for anyhow

**Comment by bengl** [comment]
@brettlangdon This also changes the error handling for cachegrind, which is #18. Basing this on `main` would mean re-doing that change upon merging #18.

However, at time of writing this comment, #18 is now merged, so this is now a non-issue.

File: src/main.rs (hunk 1 above)
Comment (bengl) [reply src/main.rs]: This is now fixed.

File: src/main.rs [hunk 2]
```diff
@@ -149,16 +151,11 @@ async fn test_timeout(timeout: u64) {
...
 #[async_std::main]
-async fn main() {
-    let config = get_config(env::args().nth(1).unwrap());
-    if config.is_err() {
-        eprintln!("{:?}", config.err().unwrap());
-        exit(1);
-    }
-    let config = config.unwrap();
+async fn main() -> Result<()> {
+    let config = get_config(env::args().nth(1).unwrap())?;
```
Comment (pawelchcki) [inline src/main.rs:155]: this `.unwrap()` is intentional?

Comment (bengl) [reply src/main.rs:155]: `.nth(1)` returns an `Option`, so I couldn't just question-mark it. That being said, this one probably warrants an `.ok_or()`. Fixing that now.

**Review by brettlangdon — APPROVED** [review]
... anyhow... lgtm

**Pull request merged**